l1-contracts/test/NewOutbox.t.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {Test} from "forge-std/Test.sol";
+import {NewOutbox} from "../src/core/messagebridge/NewOutbox.sol";
+import {INewOutbox} from "../src/core/interfaces/messagebridge/INewOutbox.sol";
+import {Errors} from "../src/core/libraries/Errors.sol";
+import {DataStructures} from "../src/core/libraries/DataStructures.sol";
+import {Merkle} from "../src/core/libraries/Merkle.sol";
+import {Hash} from "../src/core/libraries/Hash.sol";
+import {NaiveMerkle} from "./merkle/Naive.sol";
+
+contract NewOutboxTest is Test {
+  using Hash for DataStructures.L2ToL1Msg;
+
+  address internal constant _STATE_TRANSITIONER = address(0x42069123);
+  address internal constant _NOT_STATE_TRANSITIONER = address(0x69);
+  address internal constant _NOT_RECIPIENT = address(0x420);
+  uint256 internal constant _DEFAULT_TREE_HEIGHT = 2;
+  uint256 internal constant _VERSION = 0;
+
+  NewOutbox internal outbox;
+  NaiveMerkle internal zeroedTree;
+
+  function setUp() public {
+    outbox = new NewOutbox(_STATE_TRANSITIONER);
+    zeroedTree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+  }
+
+  function _fakeMessage(address _recipient) internal view returns (DataStructures.L2ToL1Msg memory) {
+    return DataStructures.L2ToL1Msg({
+      sender: DataStructures.L2Actor({
+        actor: 0x2000000000000000000000000000000000000000000000000000000000000000,
+        version: _VERSION
+      }),
+      recipient: DataStructures.L1Actor({actor: _recipient, chainId: block.chainid}),
+      content: 0x3000000000000000000000000000000000000000000000000000000000000000
+    });
+  }
+
+  function testRevertIfInsertingFromNonRollup() public {
+    bytes32 root = zeroedTree.computeRoot();
+
+    vm.prank(_NOT_STATE_TRANSITIONER);
+    vm.expectRevert(abi.encodeWithSelector(Errors.Outbox__Unauthorized.selector));
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+  }
+
+  function testRevertIfInsertingDuplicate() public {
+    bytes32 root = zeroedTree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    vm.prank(_STATE_TRANSITIONER);
+    vm.expectRevert(abi.encodeWithSelector(Errors.Outbox__RootAlreadySetAtBlock.selector, 1));
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+  }
+
+  function testInsertVariedLeafs(bytes32[] calldata _messageLeafs) public {
```

Comment:
Could you add a comment here describing what this is testing? It's not clear to me form the name.